engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/query/NadelFieldToServiceRouter.kt

```diff
@@ -0,0 +1,58 @@
+package graphql.nadel.enginekt.transform.query
+
+import graphql.nadel.Service
+import graphql.nadel.enginekt.blueprint.NadelOverallExecutionBlueprint
+import graphql.nadel.enginekt.util.copyWithChildren
+import graphql.nadel.enginekt.util.makeFieldCoordinates
+import graphql.nadel.schema.NadelDirectives
+import graphql.normalized.NormalizedField
+import graphql.normalized.NormalizedQuery
+import graphql.schema.GraphQLSchema
+
+class NadelFieldToServiceRouter {
+
+    fun execute(query: NormalizedQuery, overallExecutionBlueprint: NadelOverallExecutionBlueprint): List<FieldAndService> {
+        return query.topLevelFields.flatMap { topLevelField ->
+            when {
+                NadelNamespacedFields.isNamespacedField(topLevelField, overallExecutionBlueprint.schema) -> {
+                    val namespacedChildFieldsByService = topLevelField.children
+                        .groupBy {
+                            overallExecutionBlueprint.coordinatesToService[
```

Comment:
this should be a method inside `NadelOverallExecutionBlueprint` which takes an objecttype and a field definition